src/modules/scraper/adapter/messaging/DepositReferralConsumer.ts

```diff
@@ -48,6 +50,16 @@ export class DepositReferralConsumer {
     }
 
     await this.depositRepository.update({ id: deposit.id }, { referralAddress: referralAddress || null });
+
+    if (this.appConfig.values.stickyReferralAddressesMechanism === StickyReferralAddressesMechanism.Queue) {
+      const hasDepositsWithReferrals = await this.depositRepository.findOne({
+        where: { depositorAddr: deposit.depositorAddr, referralAddress: Not(IsNull()) },
+      });
+
+      if (hasDepositsWithReferrals) {
+        await this.depositRepository.query(updateStickyReferralAddressesForDepositor(), [deposit.depositorAddr]);
+      }
+    }
```

Comment:
If the sticky referrals are not updated by the cron, run the query which updates the referrals for this depositor only. If the depositor didn't use a referrer address at all, then don't run the query